streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -101,7 +101,9 @@ InternalTopologyBuilder builder() {
         return builder;
     }
 
-    void handleRebalanceStart() {
+    void handleRebalanceStart(final Set<String> subscribedTopics) {
+        builder.addSubscribedTopicsFromMetadata(subscribedTopics, logPrefix);
```

Comment:
This is related to the bug I found: upon assign we still need to update subscription metadata.